gensim/models/nmf.py

```diff
@@ -620,6 +604,9 @@ def update(self, corpus):
                         num_terms=self.num_tokens,
                     )
 
+                if self._W is None:
+                    self._setup(v)
```

Comment:
Nice solution!

How about a code comment, to make it clear what's going on:

```suggestion
                if self._W is None:
                    # This is the first document batch => initialize internal statistics from these initial documents.
                    # Such statistics are needed because FIXME why?
                    self._setup(v)
```

Btw what are the implications if these "first few documents" are not representative of the rest of the corpus?

Reply:
Then the `self._W` matrix might not be initialized as expected. I've changed the algorithm so that it uses the mean of the whole batch, it should be more stable.

Reply:
And if not - well, it will still work, but will take a little more time to converge.

Reply:
Added the comment.